### PR TITLE
Allow filtering of dataset by years

### PR DESCRIPTION
**Pull request by mzagaja**
This commit allows the user to filter the dataset by values available in the years column in the user interface. This also addresses #1 by removing the isMunicipal property of the model since it was merely checking to see if the table had the word “Municipal” in it to then sort by municipality.

Resolves mapc/datacommon#65

### CODE REVIEW

File: app/controllers/datasets.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default Ember.Controller.extend({
 
   download_link: Ember.computed('model', 'model.years_available.@each.selected', function() {
     let yearsSelected = this.get('model.years_available') || [];
-    if (!yearsSelected.errors) {
+    if (Ember.compare(Ember.Error, yearsSelected) !== 0) {
```
Comment (sukima) [inline app/controllers/datasets.js:20]: What is this? I've never compared an error to something before? Did you mean `yearsSelected instanceof Ember.Error`?

File: app/controllers/datasets.js [hunk 2]
```diff
@@ -76,10 +76,13 @@ export default Ember.Controller.extend({
...
   selected_rows: Ember.computed('model.years_available.@each.selected', function() {
+    if (Ember.compare(Ember.Error, this.get('model.years_available')) === 0) {
+      return this.get('model.years_available');
+    }
+    let year_column = 'row.' + this.get('model.dataset.yearcolumn');
     let years_available = this.get('model.years_available').filterBy('selected', true).map(selected => selected.year);
-    let emberModelObject = this.get('model');
     return this.get('model.raw_data.rows').filter(function(row) {
-      return years_available.includes(row.acs_year);
+      return years_available.includes(eval(year_column));
```
Comment (sukima) [inline app/controllers/datasets.js]: The coding police have been called. `eval` to lookup a value? `eval` that is not sandboxed? There is something fundamentally odd going on here.

File: app/templates/datasets.hbs [hunk 3]
```diff
@@ -107,10 +98,24 @@
...
+              <div class="content">
+                Oops.
+                <div class="sub header">
+                  We probably just need to sync and/or update permissions for this dataset.<br/><br/> We've been notified of the error and will work on it. If you'd like to contact us, we can be reached at <a href="mailto:data@mapc.org">data@mapc.org</a>
+                </div>
+              </div>
+            </h3>
+          {{else}}
+            {{ui-table model=(slice min max selected_rows) fields=model.raw_data.fields metadata=formattedMetadata}}
+            }
```
Comment (ericyoungberg) [inline app/templates/datasets.hbs]: This is a rogue bracket that renders at the bottom of the table.

File: ember-cli-build.js [hunk 4]
```diff
@@ -4,7 +4,9 @@ const EmberApp = require('ember-cli/lib/broccoli/ember-app');
 
 module.exports = function(defaults) {
   let app = new EmberApp(defaults, {
-    // Add options here
+    babel: {
+        sourceMaps: 'inline'
+      }
```
Comment (ericyoungberg) [inline ember-cli-build.js]: We don't want inline source maps for production builds. Use `EmberApp.env()` for deciding whether this value should be set or not.

**Review by ericyoungberg — CHANGES_REQUESTED** [review]
The first part of the issue is complete but there are two more things to do:

1. Only the latest year should be selected by default. The other years shouldn't be selected on initial render.
2. Except for the csv, the date range for the file links don't update as you (de)select years.

**Comment by ericyoungberg** [comment]
Also, if you attempt to deselect all year options, a `TypeError` is thrown.

**Comment by mzagaja** [comment]
@ericyoungberg Since this issue is about displaying years in the table view and not about downloads, can you file item 2 as a new issue? I am still happy to take it, but it is out of the scope of this pull request.

**Review by ericyoungberg — APPROVED** [review]

**Pull request merged**